src/main/java/sparksoniq/semantics/types/SequenceType.java

```diff
@@ -56,14 +60,23 @@ public boolean isEmptySequence() {
     }
 
     public ItemType getItemType() {
+        if (this.isEmptySequence) {
+            throw new OurBadException("Empty sequence type has no item");
+        }
         return this.itemType;
     }
 
     public Arity getArity() {
+        if (this.isEmptySequence) {
+            throw new OurBadException("Empty sequence type has no arity");
+        }
         return this.arity;
     }
 
     public boolean isSubtypeOf(SequenceType superType) {
+        if (this.isEmptySequence) {
+            return this.arity == Arity.OneOrZero || this.arity == Arity.ZeroOrMore;
+        }
```

Comment:
Empty sequence gets its arity set to null during construction. Don't these boolean expressions always evaluate to false?

Reply:
That was a mistake, it should be the superType's arity.